llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -1694,6 +1718,35 @@ let Predicates = [HasPAuth] in {
     def BLRABZ  : AuthOneOperand<0b001, 1, "blrabz">;
   }
 
+  // BLRA pseudo, generalized version of BLRAA/BLRAB/Z.
+  // This directly manipulates x16/x17, which are the only registers the OS
+  // guarantees are safe to use for sensitive operations.
```

Comment:
Unfortunately this sort of thing isn't documented.  We do describe the general contract in the security model sections of the clang docs.
For ELF it might make sense to have a formalization in the PAuth ABI, but that would have to be actually honored in an OS;  I imagine it currently isn't anywhere other than Darwin.

Reply:
Yes, having a more detailed comment here would be nice. As for now, from the comment it was unclear that this is Darwin-specific. It's also probably worth mentioning that it's harmless for non-Darwin, as you said.